vpr/src/route/binary_heap.cpp

```diff
@@ -143,12 +140,20 @@ void BinaryHeap::sift_up(size_t leaf, t_heap* const node) {
     heap_[leaf] = node;
 }
 
+//expands heap by "realloc"
 void BinaryHeap::expand_heap_if_full() {
     if (heap_tail_ > heap_size_) { /* Heap is full */
-        heap_size_ *= 2;
-        heap_ = (t_heap**)vtr::realloc((void*)(heap_ + 1),
-                                       heap_size_ * sizeof(t_heap*));
-        heap_--; /* heap goes from [1..heap_size] */
+
+        // copy elements from heap [1 ... heap_size_] to temp [0 ... heap_size_ -1]
+        std::vector<t_heap*> temp(heap_ + 1, heap_ + 1 + heap_size_);
```

Comment:
Better to do this without the temp vector. Just allocate heap_new, copy over from heap_, delete heap_, then set heap_ = heap_new. Right now there is an extra copy in a routine that is called in some cpu-intensive router code.

Bigger change (optional): could change the heap to a vector<t_heap*> and mess around with the indexing (need to keep convert between 1-based and 0-based indexing). Don't have to do that if you don't want to though, and if you do that, it should be in a separate PR.

Reply:
(This comment got lost in the original review somehow)

Reply:
Changed heap_ to a vector w/ an empty element in [0].